src/ggml-metal/ggml-metal.metal

```diff
@@ -1344,6 +1344,22 @@ kernel void kernel_ssm_scan_f32(
     }
 }
 
+kernel void kernel_argmax(
+        device   const void * x,
+        device      int32_t * dst,
+        constant    int64_t & nb01,
+        uint tgpig[[threadgroup_position_in_grid]]) {
+    device const float * x_row = (device const float *) ((device const char *) x + tgpig * nb01);
```

Comment:
We should still add `GGML_ASSERT(ggml_is_contiguous_1(src0));` because this assumes that the rows are contiguous. Also add `GGML_ASSERT(nb00 == ggml_type_size(src0->type));` since the elements in the row have to be contiguous as well.

Reply:
Done :)